samples/client/petstore/csharp-netcore/OpenAPIClient-httpclient/docs/DefaultApi.md

```diff
@@ -29,7 +30,10 @@ namespace Example
         {
             Configuration config = new Configuration();
             config.BasePath = "http://petstore.swagger.io:80/v2";
-            var apiInstance = new DefaultApi(config);
+            // create instances of HttpClient, HttpClientHandler to be reused later with different Api classes
+            HttpClient httpClient = new HttpClient();
+            HttpClientHandler httpClientHandler = new HttpClientHandler();
+            var apiInstance = new DefaultApi(httpClient, config, httpClientHandler);
```

Comment:
Hi @mandrean @frankyjuang @shibayan @Blackclaws @lucamazzanti

Do you think the following parameters order

```
config, httpClient, httpClientHandler
```

is better than we've right now as follows ?
```
httpClient, config, httpClientHandler
```

My point is should HttpClient and HttpClientHandler be appended to the constructors (that exist before the introduction of HttpClient to this generator)? That looks more natural to me but my view can be biased.

What do you guys think?

William

Reply:
Hm, for the expanded constructors its hard to tell, I'd say more people will overwrite httpClient than will overwrite config. But I do agree it looks more "natural" coming from the previous code.

Reply:
I put them in this order:
1. injected mandatory services
2. configurations and options
3. optional services

but is just a suggestion, as you want.

Reply:
👌 let's keep the current order. We can always revisit it later after we obtain more feedbacks from the community.